flink-runtime/src/test/java/org/apache/flink/runtime/highavailability/FileSystemJobResultStoreTest.java

```diff
@@ -0,0 +1,285 @@
+        assertThat(expectedCleanFile(DUMMY_JOB_RESULT_ENTRY)).exists().isFile().isNotEmpty();
+
+        assertThat(fileSystemJobResultStore.hasJobResultEntry(DUMMY_JOB_RESULT_ENTRY.getJobId()))
+                .isTrue();
+    }
+
+    @Test
+    public void testGetDirtyResultsWithNoEntry() throws IOException {
+        assertThat(fileSystemJobResultStore.getDirtyResults()).isEmpty();
+    }
+
+    @Test
+    public void testGetDirtyResultsWithDirtyEntry() throws IOException {
+        fileSystemJobResultStore.createDirtyResult(DUMMY_JOB_RESULT_ENTRY);
+        assertThat(expectedDirtyFile(DUMMY_JOB_RESULT_ENTRY)).exists().isFile().isNotEmpty();
+
+        assertThat(
+                        fileSystemJobResultStore.getDirtyResults().stream()
+                                .map(JobResult::getJobId)
+                                .collect(Collectors.toList()))
+                .containsExactlyInAnyOrder(DUMMY_JOB_RESULT_ENTRY.getJobId());
+    }
+
+    @Test
+    public void testGetDirtyResultsWithDirtyAndCleanEntry() throws IOException {
+        fileSystemJobResultStore.createDirtyResult(DUMMY_JOB_RESULT_ENTRY);
+        assertThat(expectedDirtyFile(DUMMY_JOB_RESULT_ENTRY)).exists().isFile().isNotEmpty();
+
+        fileSystemJobResultStore.markResultAsClean(DUMMY_JOB_RESULT_ENTRY.getJobId());
+        assertThat(expectedDirtyFile(DUMMY_JOB_RESULT_ENTRY)).doesNotExist();
+        assertThat(expectedCleanFile(DUMMY_JOB_RESULT_ENTRY)).exists().isFile().isNotEmpty();
+
+        final JobResultEntry dirtyJobResultEntry = createDummyJobResultEntry();
+        fileSystemJobResultStore.createDirtyResult(dirtyJobResultEntry);
+        assertThat(expectedDirtyFile(dirtyJobResultEntry)).exists().isFile().isNotEmpty();
+
+        assertThat(fileSystemJobResultStore.getDirtyResults()).hasSize(1);
+        assertThat(getCleanResultIdsFromFileSystem()).hasSize(1);
+
+        assertThat(
+                        fileSystemJobResultStore.getDirtyResults().stream()
+                                .map(JobResult::getJobId)
+                                .collect(Collectors.toList()))
+                .containsExactlyInAnyOrder(dirtyJobResultEntry.getJobId());
+    }
+
+    @Test
+    public void testDeleteOnCommit() throws IOException {
+        Path path = new Path(temporaryFolder.toURI());
+        fileSystemJobResultStore = new FileSystemJobResultStore(path.getFileSystem(), path, true);
+
+        fileSystemJobResultStore.createDirtyResult(DUMMY_JOB_RESULT_ENTRY);
+        assertThat(expectedDirtyFile(DUMMY_JOB_RESULT_ENTRY)).exists().isFile().isNotEmpty();
+
+        fileSystemJobResultStore.markResultAsClean(DUMMY_JOB_RESULT_ENTRY.getJobId());
+        assertThat(expectedDirtyFile(DUMMY_JOB_RESULT_ENTRY)).doesNotExist();
+        assertThat(expectedCleanFile(DUMMY_JOB_RESULT_ENTRY)).doesNotExist();
+    }
+
+    private static JobResultEntry createDummyJobResultEntry() {
+        return new JobResultEntry(
```

Comment:
`TestingJobResultStore` provides `DUMMY_JOB_RESULT` now in addition to `JobResult` factory methods.